"""Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT."""

import requests as requests_http
from . import utils
from sdk.models import operations, shared
from typing import Optional

class Bioentity:
    r"""Retrieval of domain entities plus associations"""
    _client: requests_http.Session
    _security_client: requests_http.Session
    _server_url: str
    _language: str
    _sdk_version: str
    _gen_version: str

    def __init__(self, client: requests_http.Session, security_client: requests_http.Session, server_url: str, language: str, sdk_version: str, gen_version: str) -> None:
        self._client = client
        self._security_client = security_client
        self._server_url = server_url
        self._language = language
        self._sdk_version = sdk_version
        self._gen_version = gen_version
        
    def get_anatomy_gene_associations(self, request: operations.GetAnatomyGeneAssociationsRequest) -> operations.GetAnatomyGeneAssociationsResponse:
        r"""Returns genes associated with a given anatomy"""
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetAnatomyGeneAssociationsRequest, base_url, '/bioentity/anatomy/{id}/genes', request)
        
        query_params = utils.get_query_params(operations.GetAnatomyGeneAssociationsRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetAnatomyGeneAssociationsResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        
        if http_res.status_code == 200:
            if utils.match_content_type(content_type, 'application/json'):
                out = utils.unmarshal_json(http_res.text, Optional[shared.AssociationResults])
                res.association_results = out

        return res

    def get_anatomy_gene_by_taxon_associations(self, request: operations.GetAnatomyGeneByTaxonAssociationsRequest) -> operations.GetAnatomyGeneByTaxonAssociationsResponse:
        r"""Returns gene IDs for all genes associated with a given anatomy, filtered by taxon
        For example, + NCBITaxon:10090 (mouse)
        """
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetAnatomyGeneByTaxonAssociationsRequest, base_url, '/bioentity/anatomy/{id}/genes/{taxid}', request)
        
        query_params = utils.get_query_params(operations.GetAnatomyGeneByTaxonAssociationsRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetAnatomyGeneByTaxonAssociationsResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        

        return res

    def get_case_disease_associations(self, request: operations.GetCaseDiseaseAssociationsRequest) -> operations.GetCaseDiseaseAssociationsResponse:
        r"""Returns diseases associated with a case"""
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetCaseDiseaseAssociationsRequest, base_url, '/bioentity/case/{id}/diseases', request)
        
        query_params = utils.get_query_params(operations.GetCaseDiseaseAssociationsRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetCaseDiseaseAssociationsResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        
        if http_res.status_code == 200:
            if utils.match_content_type(content_type, 'application/json'):
                out = utils.unmarshal_json(http_res.text, Optional[shared.AssociationResults])
                res.association_results = out

        return res

    def get_case_genotype_associations(self, request: operations.GetCaseGenotypeAssociationsRequest) -> operations.GetCaseGenotypeAssociationsResponse:
        r"""Returns genotypes associated with a case"""
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetCaseGenotypeAssociationsRequest, base_url, '/bioentity/case/{id}/genotypes', request)
        
        query_params = utils.get_query_params(operations.GetCaseGenotypeAssociationsRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetCaseGenotypeAssociationsResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        
        if http_res.status_code == 200:
            if utils.match_content_type(content_type, 'application/json'):
                out = utils.unmarshal_json(http_res.text, Optional[shared.AssociationResults])
                res.association_results = out

        return res

    def get_case_model_associations(self, request: operations.GetCaseModelAssociationsRequest) -> operations.GetCaseModelAssociationsResponse:
        r"""Returns models associated with a case"""
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetCaseModelAssociationsRequest, base_url, '/bioentity/case/{id}/models', request)
        
        query_params = utils.get_query_params(operations.GetCaseModelAssociationsRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetCaseModelAssociationsResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        
        if http_res.status_code == 200:
            if utils.match_content_type(content_type, 'application/json'):
                out = utils.unmarshal_json(http_res.text, Optional[shared.AssociationResults])
                res.association_results = out

        return res

    def get_case_phenotype_associations(self, request: operations.GetCasePhenotypeAssociationsRequest) -> operations.GetCasePhenotypeAssociationsResponse:
        r"""Returns phenotypes associated with a case"""
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetCasePhenotypeAssociationsRequest, base_url, '/bioentity/case/{id}/phenotypes', request)
        
        query_params = utils.get_query_params(operations.GetCasePhenotypeAssociationsRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetCasePhenotypeAssociationsResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        
        if http_res.status_code == 200:
            if utils.match_content_type(content_type, 'application/json'):
                out = utils.unmarshal_json(http_res.text, Optional[shared.AssociationResults])
                res.association_results = out

        return res

    def get_case_variant_associations(self, request: operations.GetCaseVariantAssociationsRequest) -> operations.GetCaseVariantAssociationsResponse:
        r"""Returns variants associated with a case"""
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetCaseVariantAssociationsRequest, base_url, '/bioentity/case/{id}/variants', request)
        
        query_params = utils.get_query_params(operations.GetCaseVariantAssociationsRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetCaseVariantAssociationsResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        
        if http_res.status_code == 200:
            if utils.match_content_type(content_type, 'application/json'):
                out = utils.unmarshal_json(http_res.text, Optional[shared.AssociationResults])
                res.association_results = out

        return res

    def get_disease_case_associations(self, request: operations.GetDiseaseCaseAssociationsRequest) -> operations.GetDiseaseCaseAssociationsResponse:
        r"""Returns cases associated with a disease"""
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetDiseaseCaseAssociationsRequest, base_url, '/bioentity/disease/{id}/cases', request)
        
        query_params = utils.get_query_params(operations.GetDiseaseCaseAssociationsRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetDiseaseCaseAssociationsResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        
        if http_res.status_code == 200:
            if utils.match_content_type(content_type, 'application/json'):
                out = utils.unmarshal_json(http_res.text, Optional[shared.AssociationResults])
                res.association_results = out

        return res

    def get_disease_gene_associations(self, request: operations.GetDiseaseGeneAssociationsRequest) -> operations.GetDiseaseGeneAssociationsResponse:
        r"""Returns genes associated with a disease"""
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetDiseaseGeneAssociationsRequest, base_url, '/bioentity/disease/{id}/genes', request)
        
        query_params = utils.get_query_params(operations.GetDiseaseGeneAssociationsRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetDiseaseGeneAssociationsResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        
        if http_res.status_code == 200:
            if utils.match_content_type(content_type, 'application/json'):
                out = utils.unmarshal_json(http_res.text, Optional[shared.AssociationResults])
                res.association_results = out

        return res

    def get_disease_genotype_associations(self, request: operations.GetDiseaseGenotypeAssociationsRequest) -> operations.GetDiseaseGenotypeAssociationsResponse:
        r"""Returns genotypes associated with a disease"""
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetDiseaseGenotypeAssociationsRequest, base_url, '/bioentity/disease/{id}/genotypes', request)
        
        query_params = utils.get_query_params(operations.GetDiseaseGenotypeAssociationsRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetDiseaseGenotypeAssociationsResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        
        if http_res.status_code == 200:
            if utils.match_content_type(content_type, 'application/json'):
                out = utils.unmarshal_json(http_res.text, Optional[shared.AssociationResults])
                res.association_results = out

        return res

    def get_disease_model_associations(self, request: operations.GetDiseaseModelAssociationsRequest) -> operations.GetDiseaseModelAssociationsResponse:
        r"""Returns associations to models of the disease
        In the association object returned, the subject will be the disease, and the object will be the model.
        The model may be a gene or genetic element.
        
        If the query disease is a general class, the association subject may be to a specific disease.
        
        In some cases the association will be *direct*, for example if a paper asserts a genotype is a model of a disease.
        
        In other cases, the association will be *indirect*, for
        example, chaining over orthology. In these cases the chain
        will be reflected in the *evidence graph*
        
        * TODO: provide hook into owlsim for dynamic computation of models by similarity
        """
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetDiseaseModelAssociationsRequest, base_url, '/bioentity/disease/{id}/models', request)
        
        query_params = utils.get_query_params(operations.GetDiseaseModelAssociationsRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetDiseaseModelAssociationsResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        
        if http_res.status_code == 200:
            if utils.match_content_type(content_type, 'application/json'):
                out = utils.unmarshal_json(http_res.text, Optional[shared.AssociationResults])
                res.association_results = out

        return res

    def get_disease_model_taxon_associations(self, request: operations.GetDiseaseModelTaxonAssociationsRequest) -> operations.GetDiseaseModelTaxonAssociationsResponse:
        r"""Returns associations to models of the disease constrained by taxon
        See /disease/<id>/models route for full details
        """
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetDiseaseModelTaxonAssociationsRequest, base_url, '/bioentity/disease/{id}/models/{taxon}', request)
        
        query_params = utils.get_query_params(operations.GetDiseaseModelTaxonAssociationsRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetDiseaseModelTaxonAssociationsResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        
        if http_res.status_code == 200:
            if utils.match_content_type(content_type, 'application/json'):
                out = utils.unmarshal_json(http_res.text, Optional[shared.AssociationResults])
                res.association_results = out

        return res

    def get_disease_pathway_associations(self, request: operations.GetDiseasePathwayAssociationsRequest) -> operations.GetDiseasePathwayAssociationsResponse:
        r"""Returns pathways associated with a disease"""
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetDiseasePathwayAssociationsRequest, base_url, '/bioentity/disease/{id}/pathways', request)
        
        query_params = utils.get_query_params(operations.GetDiseasePathwayAssociationsRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetDiseasePathwayAssociationsResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        
        if http_res.status_code == 200:
            if utils.match_content_type(content_type, 'application/json'):
                out = utils.unmarshal_json(http_res.text, Optional[shared.AssociationResults])
                res.association_results = out

        return res

    def get_disease_phenotype_associations(self, request: operations.GetDiseasePhenotypeAssociationsRequest) -> operations.GetDiseasePhenotypeAssociationsResponse:
        r"""Returns phenotypes associated with disease"""
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetDiseasePhenotypeAssociationsRequest, base_url, '/bioentity/disease/{id}/phenotypes', request)
        
        query_params = utils.get_query_params(operations.GetDiseasePhenotypeAssociationsRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetDiseasePhenotypeAssociationsResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        
        if http_res.status_code == 200:
            if utils.match_content_type(content_type, 'application/json'):
                out = utils.unmarshal_json(http_res.text, Optional[shared.D2PAssociationResults])
                res.d2_p_association_results = out

        return res

    def get_disease_publication_associations(self, request: operations.GetDiseasePublicationAssociationsRequest) -> operations.GetDiseasePublicationAssociationsResponse:
        r"""Returns publications associated with a disease"""
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetDiseasePublicationAssociationsRequest, base_url, '/bioentity/disease/{id}/publications', request)
        
        query_params = utils.get_query_params(operations.GetDiseasePublicationAssociationsRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetDiseasePublicationAssociationsResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        
        if http_res.status_code == 200:
            if utils.match_content_type(content_type, 'application/json'):
                out = utils.unmarshal_json(http_res.text, Optional[shared.AssociationResults])
                res.association_results = out

        return res

    def get_disease_substance_associations(self, request: operations.GetDiseaseSubstanceAssociationsRequest) -> operations.GetDiseaseSubstanceAssociationsResponse:
        r"""Returns substances associated with a disease
        e.g. drugs or small molecules used to treat
        """
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetDiseaseSubstanceAssociationsRequest, base_url, '/bioentity/disease/{id}/treatment', request)
        
        query_params = utils.get_query_params(operations.GetDiseaseSubstanceAssociationsRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetDiseaseSubstanceAssociationsResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        

        return res

    def get_disease_variant_associations(self, request: operations.GetDiseaseVariantAssociationsRequest) -> operations.GetDiseaseVariantAssociationsResponse:
        r"""Returns variants associated with a disease"""
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetDiseaseVariantAssociationsRequest, base_url, '/bioentity/disease/{id}/variants', request)
        
        query_params = utils.get_query_params(operations.GetDiseaseVariantAssociationsRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetDiseaseVariantAssociationsResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        
        if http_res.status_code == 200:
            if utils.match_content_type(content_type, 'application/json'):
                out = utils.unmarshal_json(http_res.text, Optional[shared.AssociationResults])
                res.association_results = out

        return res

    def get_function_associations(self, request: operations.GetFunctionAssociationsRequest) -> operations.GetFunctionAssociationsResponse:
        r"""Returns annotations associated to a function term"""
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetFunctionAssociationsRequest, base_url, '/bioentity/function/{id}', request)
        
        query_params = utils.get_query_params(operations.GetFunctionAssociationsRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetFunctionAssociationsResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        

        return res

    def get_function_gene_associations(self, request: operations.GetFunctionGeneAssociationsRequest) -> operations.GetFunctionGeneAssociationsResponse:
        r"""Returns genes associated to a GO term"""
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetFunctionGeneAssociationsRequest, base_url, '/bioentity/function/{id}/genes', request)
        
        query_params = utils.get_query_params(operations.GetFunctionGeneAssociationsRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetFunctionGeneAssociationsResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        
        if http_res.status_code == 200:
            if utils.match_content_type(content_type, 'application/json'):
                out = utils.unmarshal_json(http_res.text, Optional[shared.AssociationResults])
                res.association_results = out

        return res

    def get_function_publication_associations(self, request: operations.GetFunctionPublicationAssociationsRequest) -> operations.GetFunctionPublicationAssociationsResponse:
        r"""Returns publications associated to a GO term"""
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetFunctionPublicationAssociationsRequest, base_url, '/bioentity/function/{id}/publications', request)
        
        query_params = utils.get_query_params(operations.GetFunctionPublicationAssociationsRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetFunctionPublicationAssociationsResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        

        return res

    def get_function_taxon_associations(self, request: operations.GetFunctionTaxonAssociationsRequest) -> operations.GetFunctionTaxonAssociationsResponse:
        r"""Returns taxons associated to a GO term"""
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetFunctionTaxonAssociationsRequest, base_url, '/bioentity/function/{id}/taxons', request)
        
        query_params = utils.get_query_params(operations.GetFunctionTaxonAssociationsRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetFunctionTaxonAssociationsResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        

        return res

    def get_gene_anatomy_associations(self, request: operations.GetGeneAnatomyAssociationsRequest) -> operations.GetGeneAnatomyAssociationsResponse:
        r"""Returns anatomical entities associated with a gene"""
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetGeneAnatomyAssociationsRequest, base_url, '/bioentity/gene/{id}/anatomy', request)
        
        query_params = utils.get_query_params(operations.GetGeneAnatomyAssociationsRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetGeneAnatomyAssociationsResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        
        if http_res.status_code == 200:
            if utils.match_content_type(content_type, 'application/json'):
                out = utils.unmarshal_json(http_res.text, Optional[shared.AssociationResults])
                res.association_results = out

        return res

    def get_gene_case_associations(self, request: operations.GetGeneCaseAssociationsRequest) -> operations.GetGeneCaseAssociationsResponse:
        r"""Returns cases associated with a gene"""
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetGeneCaseAssociationsRequest, base_url, '/bioentity/gene/{id}/cases', request)
        
        query_params = utils.get_query_params(operations.GetGeneCaseAssociationsRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetGeneCaseAssociationsResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        
        if http_res.status_code == 200:
            if utils.match_content_type(content_type, 'application/json'):
                out = utils.unmarshal_json(http_res.text, Optional[shared.AssociationResults])
                res.association_results = out

        return res

    def get_gene_disease_associations(self, request: operations.GetGeneDiseaseAssociationsRequest) -> operations.GetGeneDiseaseAssociationsResponse:
        r"""Returns diseases associated with gene"""
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetGeneDiseaseAssociationsRequest, base_url, '/bioentity/gene/{id}/diseases', request)
        
        query_params = utils.get_query_params(operations.GetGeneDiseaseAssociationsRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetGeneDiseaseAssociationsResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        
        if http_res.status_code == 200:
            if utils.match_content_type(content_type, 'application/json'):
                out = utils.unmarshal_json(http_res.text, Optional[shared.AssociationResults])
                res.association_results = out

        return res

    def get_gene_expression_associations(self, request: operations.GetGeneExpressionAssociationsRequest) -> operations.GetGeneExpressionAssociationsResponse:
        r"""Returns expression events for a gene"""
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetGeneExpressionAssociationsRequest, base_url, '/bioentity/gene/{id}/expression/anatomy', request)
        
        query_params = utils.get_query_params(operations.GetGeneExpressionAssociationsRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetGeneExpressionAssociationsResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        
        if http_res.status_code == 200:
            if utils.match_content_type(content_type, 'application/json'):
                out = utils.unmarshal_json(http_res.text, Optional[shared.AssociationResults])
                res.association_results = out

        return res

    def get_gene_function_associations(self, request: operations.GetGeneFunctionAssociationsRequest) -> operations.GetGeneFunctionAssociationsResponse:
        r"""Returns function associations for a gene
        IMPLEMENTATION DETAILS
        ----------------------
        
        Note: currently this is implemented as a query to the GO/AmiGO solr instance.
        This directly supports IDs such as:
        
         - ZFIN e.g. ZFIN:ZDB-GENE-050417-357
        
        Note that the AmiGO GOlr natively stores MGI annotations to MGI:MGI:nn. However,
        the standard for biolink is MGI:nnnn, so you should use this (will be transparently
        mapped to legacy ID)
        
        Additionally, for some species such as Human, GO has the annotation attached to the UniProt ID.
        Again, this should be transparently handled; e.g. you can use NCBIGene:6469, and this will be
        mapped behind the scenes for querying.
        """
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetGeneFunctionAssociationsRequest, base_url, '/bioentity/gene/{id}/function', request)
        
        query_params = utils.get_query_params(operations.GetGeneFunctionAssociationsRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetGeneFunctionAssociationsResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        
        if http_res.status_code == 200:
            if utils.match_content_type(content_type, 'application/json'):
                out = utils.unmarshal_json(http_res.text, Optional[shared.AssociationResults])
                res.association_results = out

        return res

    def get_gene_genotype_associations(self, request: operations.GetGeneGenotypeAssociationsRequest) -> operations.GetGeneGenotypeAssociationsResponse:
        r"""Returns genotypes associated with a gene"""
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetGeneGenotypeAssociationsRequest, base_url, '/bioentity/gene/{id}/genotypes', request)
        
        query_params = utils.get_query_params(operations.GetGeneGenotypeAssociationsRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetGeneGenotypeAssociationsResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        
        if http_res.status_code == 200:
            if utils.match_content_type(content_type, 'application/json'):
                out = utils.unmarshal_json(http_res.text, Optional[shared.AssociationResults])
                res.association_results = out

        return res

    def get_gene_homolog_associations(self, request: operations.GetGeneHomologAssociationsRequest) -> operations.GetGeneHomologAssociationsResponse:
        r"""Returns homologs for a gene"""
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetGeneHomologAssociationsRequest, base_url, '/bioentity/gene/{id}/homologs', request)
        
        query_params = utils.get_query_params(operations.GetGeneHomologAssociationsRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetGeneHomologAssociationsResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        
        if http_res.status_code == 200:
            if utils.match_content_type(content_type, 'application/json'):
                out = utils.unmarshal_json(http_res.text, Optional[shared.AssociationResults])
                res.association_results = out

        return res

    def get_gene_interactions(self, request: operations.GetGeneInteractionsRequest) -> operations.GetGeneInteractionsResponse:
        r"""Returns interactions for a gene"""
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetGeneInteractionsRequest, base_url, '/bioentity/gene/{id}/interactions', request)
        
        query_params = utils.get_query_params(operations.GetGeneInteractionsRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetGeneInteractionsResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        
        if http_res.status_code == 200:
            if utils.match_content_type(content_type, 'application/json'):
                out = utils.unmarshal_json(http_res.text, Optional[shared.AssociationResults])
                res.association_results = out

        return res

    def get_gene_model_associations(self, request: operations.GetGeneModelAssociationsRequest) -> operations.GetGeneModelAssociationsResponse:
        r"""Returns models associated with a gene"""
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetGeneModelAssociationsRequest, base_url, '/bioentity/gene/{id}/models', request)
        
        query_params = utils.get_query_params(operations.GetGeneModelAssociationsRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetGeneModelAssociationsResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        
        if http_res.status_code == 200:
            if utils.match_content_type(content_type, 'application/json'):
                out = utils.unmarshal_json(http_res.text, Optional[shared.AssociationResults])
                res.association_results = out

        return res

    def get_gene_ortholog_disease_associations(self, request: operations.GetGeneOrthologDiseaseAssociationsRequest) -> operations.GetGeneOrthologDiseaseAssociationsResponse:
        r"""Return diseases associated with orthologs of a gene"""
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetGeneOrthologDiseaseAssociationsRequest, base_url, '/bioentity/gene/{id}/ortholog/diseases', request)
        
        query_params = utils.get_query_params(operations.GetGeneOrthologDiseaseAssociationsRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetGeneOrthologDiseaseAssociationsResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        
        if http_res.status_code == 200:
            if utils.match_content_type(content_type, 'application/json'):
                out = utils.unmarshal_json(http_res.text, Optional[shared.AssociationResults])
                res.association_results = out

        return res

    def get_gene_ortholog_phenotype_associations(self, request: operations.GetGeneOrthologPhenotypeAssociationsRequest) -> operations.GetGeneOrthologPhenotypeAssociationsResponse:
        r"""Return phenotypes associated with orthologs for a gene"""
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetGeneOrthologPhenotypeAssociationsRequest, base_url, '/bioentity/gene/{id}/ortholog/phenotypes', request)
        
        query_params = utils.get_query_params(operations.GetGeneOrthologPhenotypeAssociationsRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetGeneOrthologPhenotypeAssociationsResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        
        if http_res.status_code == 200:
            if utils.match_content_type(content_type, 'application/json'):
                out = utils.unmarshal_json(http_res.text, Optional[shared.AssociationResults])
                res.association_results = out

        return res

    def get_gene_pathway_associations(self, request: operations.GetGenePathwayAssociationsRequest) -> operations.GetGenePathwayAssociationsResponse:
        r"""Returns pathways associated with gene"""
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetGenePathwayAssociationsRequest, base_url, '/bioentity/gene/{id}/pathways', request)
        
        query_params = utils.get_query_params(operations.GetGenePathwayAssociationsRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetGenePathwayAssociationsResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        
        if http_res.status_code == 200:
            if utils.match_content_type(content_type, 'application/json'):
                out = utils.unmarshal_json(http_res.text, Optional[shared.AssociationResults])
                res.association_results = out

        return res

    def get_gene_phenotype_associations(self, request: operations.GetGenePhenotypeAssociationsRequest) -> operations.GetGenePhenotypeAssociationsResponse:
        r"""Returns phenotypes associated with gene"""
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetGenePhenotypeAssociationsRequest, base_url, '/bioentity/gene/{id}/phenotypes', request)
        
        query_params = utils.get_query_params(operations.GetGenePhenotypeAssociationsRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetGenePhenotypeAssociationsResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        
        if http_res.status_code == 200:
            if utils.match_content_type(content_type, 'application/json'):
                out = utils.unmarshal_json(http_res.text, Optional[shared.AssociationResults])
                res.association_results = out

        return res

    def get_gene_publication_associations(self, request: operations.GetGenePublicationAssociationsRequest) -> operations.GetGenePublicationAssociationsResponse:
        r"""Returns publications associated with a gene"""
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetGenePublicationAssociationsRequest, base_url, '/bioentity/gene/{id}/publications', request)
        
        query_params = utils.get_query_params(operations.GetGenePublicationAssociationsRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetGenePublicationAssociationsResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        
        if http_res.status_code == 200:
            if utils.match_content_type(content_type, 'application/json'):
                out = utils.unmarshal_json(http_res.text, Optional[shared.AssociationResults])
                res.association_results = out

        return res

    def get_gene_variant_associations(self, request: operations.GetGeneVariantAssociationsRequest) -> operations.GetGeneVariantAssociationsResponse:
        r"""Returns variants associated with a gene"""
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetGeneVariantAssociationsRequest, base_url, '/bioentity/gene/{id}/variants', request)
        
        query_params = utils.get_query_params(operations.GetGeneVariantAssociationsRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetGeneVariantAssociationsResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        
        if http_res.status_code == 200:
            if utils.match_content_type(content_type, 'application/json'):
                out = utils.unmarshal_json(http_res.text, Optional[shared.AssociationResults])
                res.association_results = out

        return res

    def get_generic_associations(self, request: operations.GetGenericAssociationsRequest) -> operations.GetGenericAssociationsResponse:
        r"""Returns associations for an entity regardless of the type"""
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetGenericAssociationsRequest, base_url, '/bioentity/{id}/associations', request)
        
        query_params = utils.get_query_params(operations.GetGenericAssociationsRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetGenericAssociationsResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        
        if http_res.status_code == 200:
            if utils.match_content_type(content_type, 'application/json'):
                out = utils.unmarshal_json(http_res.text, Optional[shared.AssociationResults])
                res.association_results = out

        return res

    def get_generic_object(self, request: operations.GetGenericObjectRequest) -> operations.GetGenericObjectResponse:
        r"""Returns basic info on object of any type"""
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetGenericObjectRequest, base_url, '/bioentity/{id}', request)
        
        query_params = utils.get_query_params(operations.GetGenericObjectRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetGenericObjectResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        
        if http_res.status_code == 200:
            if utils.match_content_type(content_type, 'application/json'):
                out = utils.unmarshal_json(http_res.text, Optional[shared.BioObject])
                res.bio_object = out

        return res

    def get_generic_object_by_type(self, request: operations.GetGenericObjectByTypeRequest) -> operations.GetGenericObjectByTypeResponse:
        r"""Return basic info on an object for a given type"""
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetGenericObjectByTypeRequest, base_url, '/bioentity/{type}/{id}', request)
        
        query_params = utils.get_query_params(operations.GetGenericObjectByTypeRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetGenericObjectByTypeResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        

        return res

    def get_genotype_case_associations(self, request: operations.GetGenotypeCaseAssociationsRequest) -> operations.GetGenotypeCaseAssociationsResponse:
        r"""Returns cases associated with a genotype"""
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetGenotypeCaseAssociationsRequest, base_url, '/bioentity/genotype/{id}/cases', request)
        
        query_params = utils.get_query_params(operations.GetGenotypeCaseAssociationsRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetGenotypeCaseAssociationsResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        
        if http_res.status_code == 200:
            if utils.match_content_type(content_type, 'application/json'):
                out = utils.unmarshal_json(http_res.text, Optional[shared.AssociationResults])
                res.association_results = out

        return res

    def get_genotype_disease_associations(self, request: operations.GetGenotypeDiseaseAssociationsRequest) -> operations.GetGenotypeDiseaseAssociationsResponse:
        r"""Returns diseases associated with a genotype"""
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetGenotypeDiseaseAssociationsRequest, base_url, '/bioentity/genotype/{id}/diseases', request)
        
        query_params = utils.get_query_params(operations.GetGenotypeDiseaseAssociationsRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetGenotypeDiseaseAssociationsResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        
        if http_res.status_code == 200:
            if utils.match_content_type(content_type, 'application/json'):
                out = utils.unmarshal_json(http_res.text, Optional[shared.AssociationResults])
                res.association_results = out

        return res

    def get_genotype_gene_associations(self, request: operations.GetGenotypeGeneAssociationsRequest) -> operations.GetGenotypeGeneAssociationsResponse:
        r"""Returns genes associated with a genotype"""
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetGenotypeGeneAssociationsRequest, base_url, '/bioentity/genotype/{id}/genes', request)
        
        query_params = utils.get_query_params(operations.GetGenotypeGeneAssociationsRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetGenotypeGeneAssociationsResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        
        if http_res.status_code == 200:
            if utils.match_content_type(content_type, 'application/json'):
                out = utils.unmarshal_json(http_res.text, Optional[shared.AssociationResults])
                res.association_results = out

        return res

    def get_genotype_genotype_associations(self, request: operations.GetGenotypeGenotypeAssociationsRequest) -> operations.GetGenotypeGenotypeAssociationsResponse:
        r"""Returns genotypes-genotype associations
        Genotypes may be related to one another according to the GENO model
        """
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetGenotypeGenotypeAssociationsRequest, base_url, '/bioentity/genotype/{id}/genotypes', request)
        
        query_params = utils.get_query_params(operations.GetGenotypeGenotypeAssociationsRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetGenotypeGenotypeAssociationsResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        
        if http_res.status_code == 200:
            if utils.match_content_type(content_type, 'application/json'):
                out = utils.unmarshal_json(http_res.text, Optional[shared.AssociationResults])
                res.association_results = out

        return res

    def get_genotype_model_associations(self, request: operations.GetGenotypeModelAssociationsRequest) -> operations.GetGenotypeModelAssociationsResponse:
        r"""Returns models associated with a genotype"""
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetGenotypeModelAssociationsRequest, base_url, '/bioentity/genotype/{id}/models', request)
        
        query_params = utils.get_query_params(operations.GetGenotypeModelAssociationsRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetGenotypeModelAssociationsResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        
        if http_res.status_code == 200:
            if utils.match_content_type(content_type, 'application/json'):
                out = utils.unmarshal_json(http_res.text, Optional[shared.AssociationResults])
                res.association_results = out

        return res

    def get_genotype_phenotype_associations(self, request: operations.GetGenotypePhenotypeAssociationsRequest) -> operations.GetGenotypePhenotypeAssociationsResponse:
        r"""Returns phenotypes associated with a genotype"""
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetGenotypePhenotypeAssociationsRequest, base_url, '/bioentity/genotype/{id}/phenotypes', request)
        
        query_params = utils.get_query_params(operations.GetGenotypePhenotypeAssociationsRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetGenotypePhenotypeAssociationsResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        
        if http_res.status_code == 200:
            if utils.match_content_type(content_type, 'application/json'):
                out = utils.unmarshal_json(http_res.text, Optional[shared.AssociationResults])
                res.association_results = out

        return res

    def get_genotype_publication_associations(self, request: operations.GetGenotypePublicationAssociationsRequest) -> operations.GetGenotypePublicationAssociationsResponse:
        r"""Returns publications associated with a genotype"""
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetGenotypePublicationAssociationsRequest, base_url, '/bioentity/genotype/{id}/publications', request)
        
        query_params = utils.get_query_params(operations.GetGenotypePublicationAssociationsRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetGenotypePublicationAssociationsResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        
        if http_res.status_code == 200:
            if utils.match_content_type(content_type, 'application/json'):
                out = utils.unmarshal_json(http_res.text, Optional[shared.AssociationResults])
                res.association_results = out

        return res

    def get_genotype_variant_associations(self, request: operations.GetGenotypeVariantAssociationsRequest) -> operations.GetGenotypeVariantAssociationsResponse:
        r"""Returns genotypes-variant associations"""
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetGenotypeVariantAssociationsRequest, base_url, '/bioentity/genotype/{id}/variants', request)
        
        query_params = utils.get_query_params(operations.GetGenotypeVariantAssociationsRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetGenotypeVariantAssociationsResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        
        if http_res.status_code == 200:
            if utils.match_content_type(content_type, 'application/json'):
                out = utils.unmarshal_json(http_res.text, Optional[shared.AssociationResults])
                res.association_results = out

        return res

    def get_goterm_gene_associations(self, request: operations.GetGotermGeneAssociationsRequest) -> operations.GetGotermGeneAssociationsResponse:
        r"""Returns associations to GO terms for a gene"""
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetGotermGeneAssociationsRequest, base_url, '/bioentity/goterm/{id}/genes', request)
        
        query_params = utils.get_query_params(operations.GetGotermGeneAssociationsRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetGotermGeneAssociationsResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        
        if http_res.status_code == 200:
            if utils.match_content_type(content_type, 'application/json'):
                out = utils.unmarshal_json(http_res.text, Optional[shared.AssociationResults])
                res.association_results = out

        return res

    def get_model_case_associations(self, request: operations.GetModelCaseAssociationsRequest) -> operations.GetModelCaseAssociationsResponse:
        r"""Returns cases associated with a model"""
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetModelCaseAssociationsRequest, base_url, '/bioentity/model/{id}/cases', request)
        
        query_params = utils.get_query_params(operations.GetModelCaseAssociationsRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetModelCaseAssociationsResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        
        if http_res.status_code == 200:
            if utils.match_content_type(content_type, 'application/json'):
                out = utils.unmarshal_json(http_res.text, Optional[shared.AssociationResults])
                res.association_results = out

        return res

    def get_model_disease_associations(self, request: operations.GetModelDiseaseAssociationsRequest) -> operations.GetModelDiseaseAssociationsResponse:
        r"""Returns diseases associated with a model"""
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetModelDiseaseAssociationsRequest, base_url, '/bioentity/model/{id}/diseases', request)
        
        query_params = utils.get_query_params(operations.GetModelDiseaseAssociationsRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetModelDiseaseAssociationsResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        
        if http_res.status_code == 200:
            if utils.match_content_type(content_type, 'application/json'):
                out = utils.unmarshal_json(http_res.text, Optional[shared.AssociationResults])
                res.association_results = out

        return res

    def get_model_gene_associations(self, request: operations.GetModelGeneAssociationsRequest) -> operations.GetModelGeneAssociationsResponse:
        r"""Returns genes associated with a model"""
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetModelGeneAssociationsRequest, base_url, '/bioentity/model/{id}/genes', request)
        
        query_params = utils.get_query_params(operations.GetModelGeneAssociationsRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetModelGeneAssociationsResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        
        if http_res.status_code == 200:
            if utils.match_content_type(content_type, 'application/json'):
                out = utils.unmarshal_json(http_res.text, Optional[shared.AssociationResults])
                res.association_results = out

        return res

    def get_model_genotype_associations(self, request: operations.GetModelGenotypeAssociationsRequest) -> operations.GetModelGenotypeAssociationsResponse:
        r"""Returns genotypes associated with a model"""
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetModelGenotypeAssociationsRequest, base_url, '/bioentity/model/{id}/genotypes', request)
        
        query_params = utils.get_query_params(operations.GetModelGenotypeAssociationsRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetModelGenotypeAssociationsResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        
        if http_res.status_code == 200:
            if utils.match_content_type(content_type, 'application/json'):
                out = utils.unmarshal_json(http_res.text, Optional[shared.AssociationResults])
                res.association_results = out

        return res

    def get_model_phenotype_associations(self, request: operations.GetModelPhenotypeAssociationsRequest) -> operations.GetModelPhenotypeAssociationsResponse:
        r"""Returns phenotypes associated with a model"""
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetModelPhenotypeAssociationsRequest, base_url, '/bioentity/model/{id}/phenotypes', request)
        
        query_params = utils.get_query_params(operations.GetModelPhenotypeAssociationsRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetModelPhenotypeAssociationsResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        
        if http_res.status_code == 200:
            if utils.match_content_type(content_type, 'application/json'):
                out = utils.unmarshal_json(http_res.text, Optional[shared.AssociationResults])
                res.association_results = out

        return res

    def get_model_publication_associations(self, request: operations.GetModelPublicationAssociationsRequest) -> operations.GetModelPublicationAssociationsResponse:
        r"""Returns publications associated with a model"""
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetModelPublicationAssociationsRequest, base_url, '/bioentity/model/{id}/publications', request)
        
        query_params = utils.get_query_params(operations.GetModelPublicationAssociationsRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetModelPublicationAssociationsResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        
        if http_res.status_code == 200:
            if utils.match_content_type(content_type, 'application/json'):
                out = utils.unmarshal_json(http_res.text, Optional[shared.AssociationResults])
                res.association_results = out

        return res

    def get_model_variant_associations(self, request: operations.GetModelVariantAssociationsRequest) -> operations.GetModelVariantAssociationsResponse:
        r"""Returns variants associated with a model"""
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetModelVariantAssociationsRequest, base_url, '/bioentity/model/{id}/variants', request)
        
        query_params = utils.get_query_params(operations.GetModelVariantAssociationsRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetModelVariantAssociationsResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        
        if http_res.status_code == 200:
            if utils.match_content_type(content_type, 'application/json'):
                out = utils.unmarshal_json(http_res.text, Optional[shared.AssociationResults])
                res.association_results = out

        return res

    def get_pathway_disease_associations(self, request: operations.GetPathwayDiseaseAssociationsRequest) -> operations.GetPathwayDiseaseAssociationsResponse:
        r"""Returns diseases associated with a pathway"""
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetPathwayDiseaseAssociationsRequest, base_url, '/bioentity/pathway/{id}/diseases', request)
        
        query_params = utils.get_query_params(operations.GetPathwayDiseaseAssociationsRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetPathwayDiseaseAssociationsResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        
        if http_res.status_code == 200:
            if utils.match_content_type(content_type, 'application/json'):
                out = utils.unmarshal_json(http_res.text, Optional[shared.AssociationResults])
                res.association_results = out

        return res

    def get_pathway_gene_associations(self, request: operations.GetPathwayGeneAssociationsRequest) -> operations.GetPathwayGeneAssociationsResponse:
        r"""Returns genes associated with a pathway"""
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetPathwayGeneAssociationsRequest, base_url, '/bioentity/pathway/{id}/genes', request)
        
        query_params = utils.get_query_params(operations.GetPathwayGeneAssociationsRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetPathwayGeneAssociationsResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        
        if http_res.status_code == 200:
            if utils.match_content_type(content_type, 'application/json'):
                out = utils.unmarshal_json(http_res.text, Optional[shared.AssociationResults])
                res.association_results = out

        return res

    def get_pathway_phenotype_associations(self, request: operations.GetPathwayPhenotypeAssociationsRequest) -> operations.GetPathwayPhenotypeAssociationsResponse:
        r"""Returns phenotypes associated with a pathway"""
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetPathwayPhenotypeAssociationsRequest, base_url, '/bioentity/pathway/{id}/phenotypes', request)
        
        query_params = utils.get_query_params(operations.GetPathwayPhenotypeAssociationsRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetPathwayPhenotypeAssociationsResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        
        if http_res.status_code == 200:
            if utils.match_content_type(content_type, 'application/json'):
                out = utils.unmarshal_json(http_res.text, Optional[shared.AssociationResults])
                res.association_results = out

        return res

    def get_phenotype_anatomy_associations(self, request: operations.GetPhenotypeAnatomyAssociationsRequest) -> operations.GetPhenotypeAnatomyAssociationsResponse:
        r"""Returns anatomical entities associated with a phenotype
        Example IDs:
        
         * MP:0008521 abnormal Bowman membrane
        """
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetPhenotypeAnatomyAssociationsRequest, base_url, '/bioentity/phenotype/{id}/anatomy', request)
        
        query_params = utils.get_query_params(operations.GetPhenotypeAnatomyAssociationsRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetPhenotypeAnatomyAssociationsResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        
        if http_res.status_code == 200:
            if utils.match_content_type(content_type, 'application/json'):
                out = utils.unmarshal_json(http_res.text, Optional[list[shared.NamedObject]])
                res.named_objects = out

        return res

    def get_phenotype_case_associations(self, request: operations.GetPhenotypeCaseAssociationsRequest) -> operations.GetPhenotypeCaseAssociationsResponse:
        r"""Returns cases associated with a phenotype"""
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetPhenotypeCaseAssociationsRequest, base_url, '/bioentity/phenotype/{id}/cases', request)
        
        query_params = utils.get_query_params(operations.GetPhenotypeCaseAssociationsRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetPhenotypeCaseAssociationsResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        
        if http_res.status_code == 200:
            if utils.match_content_type(content_type, 'application/json'):
                out = utils.unmarshal_json(http_res.text, Optional[shared.AssociationResults])
                res.association_results = out

        return res

    def get_phenotype_disease_associations(self, request: operations.GetPhenotypeDiseaseAssociationsRequest) -> operations.GetPhenotypeDiseaseAssociationsResponse:
        r"""Returns diseases associated with a phenotype"""
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetPhenotypeDiseaseAssociationsRequest, base_url, '/bioentity/phenotype/{id}/diseases', request)
        
        query_params = utils.get_query_params(operations.GetPhenotypeDiseaseAssociationsRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetPhenotypeDiseaseAssociationsResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        
        if http_res.status_code == 200:
            if utils.match_content_type(content_type, 'application/json'):
                out = utils.unmarshal_json(http_res.text, Optional[shared.D2PAssociationResults])
                res.d2_p_association_results = out

        return res

    def get_phenotype_gene_associations(self, request: operations.GetPhenotypeGeneAssociationsRequest) -> operations.GetPhenotypeGeneAssociationsResponse:
        r"""Returns genes associated with a phenotype"""
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetPhenotypeGeneAssociationsRequest, base_url, '/bioentity/phenotype/{id}/genes', request)
        
        query_params = utils.get_query_params(operations.GetPhenotypeGeneAssociationsRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetPhenotypeGeneAssociationsResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        
        if http_res.status_code == 200:
            if utils.match_content_type(content_type, 'application/json'):
                out = utils.unmarshal_json(http_res.text, Optional[shared.AssociationResults])
                res.association_results = out

        return res

    def get_phenotype_gene_by_taxon_associations(self, request: operations.GetPhenotypeGeneByTaxonAssociationsRequest) -> operations.GetPhenotypeGeneByTaxonAssociationsResponse:
        r"""Returns gene IDs for all genes associated with a given phenotype, filtered by taxon
        For example, MP:0001569 + NCBITaxon:10090 (mouse)
        """
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetPhenotypeGeneByTaxonAssociationsRequest, base_url, '/bioentity/phenotype/{id}/gene/{taxid}/ids', request)
        
        query_params = utils.get_query_params(operations.GetPhenotypeGeneByTaxonAssociationsRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetPhenotypeGeneByTaxonAssociationsResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        

        return res

    def get_phenotype_genotype_associations(self, request: operations.GetPhenotypeGenotypeAssociationsRequest) -> operations.GetPhenotypeGenotypeAssociationsResponse:
        r"""Returns genotypes associated with a phenotype"""
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetPhenotypeGenotypeAssociationsRequest, base_url, '/bioentity/phenotype/{id}/genotypes', request)
        
        query_params = utils.get_query_params(operations.GetPhenotypeGenotypeAssociationsRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetPhenotypeGenotypeAssociationsResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        
        if http_res.status_code == 200:
            if utils.match_content_type(content_type, 'application/json'):
                out = utils.unmarshal_json(http_res.text, Optional[shared.AssociationResults])
                res.association_results = out

        return res

    def get_phenotype_pathway_associations(self, request: operations.GetPhenotypePathwayAssociationsRequest) -> operations.GetPhenotypePathwayAssociationsResponse:
        r"""Returns pathways associated with a phenotype"""
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetPhenotypePathwayAssociationsRequest, base_url, '/bioentity/phenotype/{id}/pathways', request)
        
        query_params = utils.get_query_params(operations.GetPhenotypePathwayAssociationsRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetPhenotypePathwayAssociationsResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        
        if http_res.status_code == 200:
            if utils.match_content_type(content_type, 'application/json'):
                out = utils.unmarshal_json(http_res.text, Optional[shared.AssociationResults])
                res.association_results = out

        return res

    def get_phenotype_publication_associations(self, request: operations.GetPhenotypePublicationAssociationsRequest) -> operations.GetPhenotypePublicationAssociationsResponse:
        r"""Returns publications associated with a phenotype"""
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetPhenotypePublicationAssociationsRequest, base_url, '/bioentity/phenotype/{id}/publications', request)
        
        query_params = utils.get_query_params(operations.GetPhenotypePublicationAssociationsRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetPhenotypePublicationAssociationsResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        
        if http_res.status_code == 200:
            if utils.match_content_type(content_type, 'application/json'):
                out = utils.unmarshal_json(http_res.text, Optional[shared.AssociationResults])
                res.association_results = out

        return res

    def get_phenotype_variant_associations(self, request: operations.GetPhenotypeVariantAssociationsRequest) -> operations.GetPhenotypeVariantAssociationsResponse:
        r"""Returns variants associated with a phenotype"""
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetPhenotypeVariantAssociationsRequest, base_url, '/bioentity/phenotype/{id}/variants', request)
        
        query_params = utils.get_query_params(operations.GetPhenotypeVariantAssociationsRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetPhenotypeVariantAssociationsResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        
        if http_res.status_code == 200:
            if utils.match_content_type(content_type, 'application/json'):
                out = utils.unmarshal_json(http_res.text, Optional[shared.AssociationResults])
                res.association_results = out

        return res

    def get_publication_disease_associations(self, request: operations.GetPublicationDiseaseAssociationsRequest) -> operations.GetPublicationDiseaseAssociationsResponse:
        r"""Returns diseases associated with a publication"""
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetPublicationDiseaseAssociationsRequest, base_url, '/bioentity/publication/{id}/diseases', request)
        
        query_params = utils.get_query_params(operations.GetPublicationDiseaseAssociationsRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetPublicationDiseaseAssociationsResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        
        if http_res.status_code == 200:
            if utils.match_content_type(content_type, 'application/json'):
                out = utils.unmarshal_json(http_res.text, Optional[shared.AssociationResults])
                res.association_results = out

        return res

    def get_publication_gene_associations(self, request: operations.GetPublicationGeneAssociationsRequest) -> operations.GetPublicationGeneAssociationsResponse:
        r"""Returns genes associated with a publication"""
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetPublicationGeneAssociationsRequest, base_url, '/bioentity/publication/{id}/genes', request)
        
        query_params = utils.get_query_params(operations.GetPublicationGeneAssociationsRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetPublicationGeneAssociationsResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        
        if http_res.status_code == 200:
            if utils.match_content_type(content_type, 'application/json'):
                out = utils.unmarshal_json(http_res.text, Optional[shared.AssociationResults])
                res.association_results = out

        return res

    def get_publication_genotype_associations(self, request: operations.GetPublicationGenotypeAssociationsRequest) -> operations.GetPublicationGenotypeAssociationsResponse:
        r"""Returns genotypes associated with a publication"""
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetPublicationGenotypeAssociationsRequest, base_url, '/bioentity/publication/{id}/genotypes', request)
        
        query_params = utils.get_query_params(operations.GetPublicationGenotypeAssociationsRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetPublicationGenotypeAssociationsResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        
        if http_res.status_code == 200:
            if utils.match_content_type(content_type, 'application/json'):
                out = utils.unmarshal_json(http_res.text, Optional[shared.AssociationResults])
                res.association_results = out

        return res

    def get_publication_model_associations(self, request: operations.GetPublicationModelAssociationsRequest) -> operations.GetPublicationModelAssociationsResponse:
        r"""Returns models associated with a publication"""
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetPublicationModelAssociationsRequest, base_url, '/bioentity/publication/{id}/models', request)
        
        query_params = utils.get_query_params(operations.GetPublicationModelAssociationsRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetPublicationModelAssociationsResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        
        if http_res.status_code == 200:
            if utils.match_content_type(content_type, 'application/json'):
                out = utils.unmarshal_json(http_res.text, Optional[shared.AssociationResults])
                res.association_results = out

        return res

    def get_publication_phenotype_associations(self, request: operations.GetPublicationPhenotypeAssociationsRequest) -> operations.GetPublicationPhenotypeAssociationsResponse:
        r"""Returns phenotypes associated with a publication"""
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetPublicationPhenotypeAssociationsRequest, base_url, '/bioentity/publication/{id}/phenotypes', request)
        
        query_params = utils.get_query_params(operations.GetPublicationPhenotypeAssociationsRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetPublicationPhenotypeAssociationsResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        
        if http_res.status_code == 200:
            if utils.match_content_type(content_type, 'application/json'):
                out = utils.unmarshal_json(http_res.text, Optional[shared.AssociationResults])
                res.association_results = out

        return res

    def get_publication_variant_associations(self, request: operations.GetPublicationVariantAssociationsRequest) -> operations.GetPublicationVariantAssociationsResponse:
        r"""Returns variants associated with a publication"""
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetPublicationVariantAssociationsRequest, base_url, '/bioentity/publication/{id}/variants', request)
        
        query_params = utils.get_query_params(operations.GetPublicationVariantAssociationsRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetPublicationVariantAssociationsResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        
        if http_res.status_code == 200:
            if utils.match_content_type(content_type, 'application/json'):
                out = utils.unmarshal_json(http_res.text, Optional[shared.AssociationResults])
                res.association_results = out

        return res

    def get_substance_participant_in_associations(self, request: operations.GetSubstanceParticipantInAssociationsRequest) -> operations.GetSubstanceParticipantInAssociationsResponse:
        r"""Returns associations between an activity and process and the specified substance
        Examples relationships:
        
         * substance is a metabolite of a process
         * substance is synthesized by a process
         * substance is modified by an activity
         * substance elicits a response program/pathway
         * substance is transported by activity or pathway
        
        For example, CHEBI:40036 (amitrole)
        """
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetSubstanceParticipantInAssociationsRequest, base_url, '/bioentity/substance/{id}/participant_in', request)
        
        query_params = utils.get_query_params(operations.GetSubstanceParticipantInAssociationsRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetSubstanceParticipantInAssociationsResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        
        if http_res.status_code == 200:
            if utils.match_content_type(content_type, 'application/json'):
                out = utils.unmarshal_json(http_res.text, Optional[list[shared.Association]])
                res.associations = out

        return res

    def get_substance_role_associations(self, request: operations.GetSubstanceRoleAssociationsRequest) -> operations.GetSubstanceRoleAssociationsResponse:
        r"""Returns associations between given drug and roles
        Roles may be human-oriented (e.g. pesticide) or molecular (e.g. enzyme inhibitor)
        """
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetSubstanceRoleAssociationsRequest, base_url, '/bioentity/substance/{id}/roles', request)
        
        query_params = utils.get_query_params(operations.GetSubstanceRoleAssociationsRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetSubstanceRoleAssociationsResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        
        if http_res.status_code == 200:
            if utils.match_content_type(content_type, 'application/json'):
                out = utils.unmarshal_json(http_res.text, Optional[list[shared.Association]])
                res.associations = out

        return res

    def get_substance_treats_associations(self, request: operations.GetSubstanceTreatsAssociationsRequest) -> operations.GetSubstanceTreatsAssociationsResponse:
        r"""Returns substances associated with a disease
        e.g. drugs or small molecules used to treat
        """
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetSubstanceTreatsAssociationsRequest, base_url, '/bioentity/substance/{id}/treats', request)
        
        query_params = utils.get_query_params(operations.GetSubstanceTreatsAssociationsRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetSubstanceTreatsAssociationsResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        

        return res

    def get_variant_case_associations(self, request: operations.GetVariantCaseAssociationsRequest) -> operations.GetVariantCaseAssociationsResponse:
        r"""Returns cases associated with a variant"""
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetVariantCaseAssociationsRequest, base_url, '/bioentity/variant/{id}/cases', request)
        
        query_params = utils.get_query_params(operations.GetVariantCaseAssociationsRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetVariantCaseAssociationsResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        
        if http_res.status_code == 200:
            if utils.match_content_type(content_type, 'application/json'):
                out = utils.unmarshal_json(http_res.text, Optional[shared.AssociationResults])
                res.association_results = out

        return res

    def get_variant_disease_associations(self, request: operations.GetVariantDiseaseAssociationsRequest) -> operations.GetVariantDiseaseAssociationsResponse:
        r"""Returns diseases associated with a variant"""
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetVariantDiseaseAssociationsRequest, base_url, '/bioentity/variant/{id}/diseases', request)
        
        query_params = utils.get_query_params(operations.GetVariantDiseaseAssociationsRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetVariantDiseaseAssociationsResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        
        if http_res.status_code == 200:
            if utils.match_content_type(content_type, 'application/json'):
                out = utils.unmarshal_json(http_res.text, Optional[shared.AssociationResults])
                res.association_results = out

        return res

    def get_variant_gene_associations(self, request: operations.GetVariantGeneAssociationsRequest) -> operations.GetVariantGeneAssociationsResponse:
        r"""Returns genes associated with a variant"""
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetVariantGeneAssociationsRequest, base_url, '/bioentity/variant/{id}/genes', request)
        
        query_params = utils.get_query_params(operations.GetVariantGeneAssociationsRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetVariantGeneAssociationsResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        
        if http_res.status_code == 200:
            if utils.match_content_type(content_type, 'application/json'):
                out = utils.unmarshal_json(http_res.text, Optional[shared.AssociationResults])
                res.association_results = out

        return res

    def get_variant_genotype_associations(self, request: operations.GetVariantGenotypeAssociationsRequest) -> operations.GetVariantGenotypeAssociationsResponse:
        r"""Returns genotypes associated with a variant"""
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetVariantGenotypeAssociationsRequest, base_url, '/bioentity/variant/{id}/genotypes', request)
        
        query_params = utils.get_query_params(operations.GetVariantGenotypeAssociationsRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetVariantGenotypeAssociationsResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        
        if http_res.status_code == 200:
            if utils.match_content_type(content_type, 'application/json'):
                out = utils.unmarshal_json(http_res.text, Optional[shared.AssociationResults])
                res.association_results = out

        return res

    def get_variant_model_associations(self, request: operations.GetVariantModelAssociationsRequest) -> operations.GetVariantModelAssociationsResponse:
        r"""Returns models associated with a variant"""
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetVariantModelAssociationsRequest, base_url, '/bioentity/variant/{id}/models', request)
        
        query_params = utils.get_query_params(operations.GetVariantModelAssociationsRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetVariantModelAssociationsResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        
        if http_res.status_code == 200:
            if utils.match_content_type(content_type, 'application/json'):
                out = utils.unmarshal_json(http_res.text, Optional[shared.AssociationResults])
                res.association_results = out

        return res

    def get_variant_phenotype_associations(self, request: operations.GetVariantPhenotypeAssociationsRequest) -> operations.GetVariantPhenotypeAssociationsResponse:
        r"""Returns phenotypes associated with a variant"""
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetVariantPhenotypeAssociationsRequest, base_url, '/bioentity/variant/{id}/phenotypes', request)
        
        query_params = utils.get_query_params(operations.GetVariantPhenotypeAssociationsRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetVariantPhenotypeAssociationsResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        
        if http_res.status_code == 200:
            if utils.match_content_type(content_type, 'application/json'):
                out = utils.unmarshal_json(http_res.text, Optional[shared.AssociationResults])
                res.association_results = out

        return res

    def get_variant_publication_associations(self, request: operations.GetVariantPublicationAssociationsRequest) -> operations.GetVariantPublicationAssociationsResponse:
        r"""Returns publications associated with a variant"""
        base_url = self._server_url
        
        url = utils.generate_url(operations.GetVariantPublicationAssociationsRequest, base_url, '/bioentity/variant/{id}/publications', request)
        
        query_params = utils.get_query_params(operations.GetVariantPublicationAssociationsRequest, request)
        
        client = self._client
        
        http_res = client.request('GET', url, params=query_params)
        content_type = http_res.headers.get('Content-Type')

        res = operations.GetVariantPublicationAssociationsResponse(status_code=http_res.status_code, content_type=content_type, raw_response=http_res)
        
        if http_res.status_code == 200:
            if utils.match_content_type(content_type, 'application/json'):
                out = utils.unmarshal_json(http_res.text, Optional[shared.AssociationResults])
                res.association_results = out

        return res

    